internal/pkg/addon/errors.go

```diff
@@ -10,13 +10,13 @@ import (
 	"gopkg.in/yaml.v3"
 )
 
-// ErrAddonsDirNotExist occurs when an addons directory for a workload does not exist.
-type ErrAddonsDirNotExist struct {
+// ErrAddonsNotFound occurs when an addons directory for a workload is either not found or empty.
+type ErrAddonsNotFound struct {
 	WlName    string
 	ParentErr error
 }
 
-func (e *ErrAddonsDirNotExist) Error() string {
+func (e *ErrAddonsNotFound) Error() string {
 	return fmt.Sprintf("read addons directory for %s: %v", e.WlName, e.ParentErr)
```

Comment:
nit: could we include a conditional so that this returns a different error message if `e.ParrentErr == nil`? For example,
```suggestion
        if e.ParrentErr != nil {
            return fmt.Sprintf("read addons directory for %s: %v", e.WlName, e.ParentErr)
        }
        return fmt.Sprintf("read addons directory for %s: no addons found", e.WlName)

```

Reply:
Great idea, will update this shortly. Thanks @Lou1415926 🙏🏾

Reply:
@Lou1415926 , This is the first error I see with some formatting logic. Should we include a test case for this?

Reply:
@buddyspike Yay thanks for asking!! Currently we don't include unit tests for custom errors' formatting logic anywhere, but I think it'd be great to have them in the future. That'd be a refactoring that we can do on our end though. I think it's fine for this PR to go without it 🚢.

Reply:
Awesome! I asked the question after I wrote the test lolz. I didn't include it in my patch but I'll post it here anyway 😊. 

```
package addon

import (
	"errors"
	"fmt"
	"github.com/stretchr/testify/require"
	"testing"
)

func TestErrors(t *testing.T) {
	var (
		wl        = "mysvc"
		parentErr = errors.New("some err")
	)

	testCases := map[string]struct {
		error     func() error
		wantedMsg string
	}{
		"returns a message including parent error": {
			error: func() error {
				return &ErrAddonsNotFound{
					WlName:    wl,
					ParentErr: parentErr,
				}
			},
			wantedMsg: fmt.Sprintf("read addons directory for %s: %v", wl, parentErr),
		},
		"returns no addons found message": {
			error: func() error {
				return &ErrAddonsNotFound{
					WlName: wl,
				}
			},
			wantedMsg: fmt.Sprintf("read addons directory for %s: no addons found", wl),
		},
	}

	for name, tc := range testCases {
		t.Run(name, func(t *testing.T) {
			// GIVEN
			err := tc.error()

			// WHEN
			msg := err.Error()

			// THEN
			require.Equal(t, tc.wantedMsg, msg)
		})
	}
}

```

Reply:
this is awesome!! thanks ❤️